ndcube/mixins/ndcubeio.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+import numpy as np
+from astropy.nddata import NDIOMixin, CCDData
+from astropy.io import fits, registry
+from astropy import units as u
+from astropy.nddata.nduncertainty import (
+    StdDevUncertainty, NDUncertainty, VarianceUncertainty, InverseVariance)
+
+_known_uncertainties = (StdDevUncertainty, VarianceUncertainty, InverseVariance)
+_unc_name_to_cls = {cls.__name__: cls for cls in _known_uncertainties}
+_unc_cls_to_name = {cls: cls.__name__ for cls in _known_uncertainties}
+
+
+__all__ = ['NDCubeIOMixin']
+
+# This file is written to define a mixin class to support
+# write option in ndcube. Uses `astropy.io.registry`
+# to register the write function
+
+
+def flatten(lst):
+    """Helper function to flatten a list of dictionary,
+       resulted out of the form :
+       {Key1:Value1, {Key2:Value2, Key3:Value3}}
+
+
+    Parameters
+    ----------
+    lst : list
+        list of dictionary
+
+    Returns
+    -------
+    list
+        list of flattened list
+    Notes
+    -----
+    Format of the output:
+    [{Key1:Value1, Key2:Value2, Key3:Value3}]
+    """
+
+    res1 = list()
+    for entry in lst:
+        for key, value in entry.items():
+            res = dict()
+            res['name'] = key
+            for k,v in value.items():
+                res[k] = v
+        res1.append(res)
+    return res1
+
+
+
+def _insert_in_metadata_fits_safe(meta, key, value):
+    """Helper function to insert key-value pair into metadata in a way that
+       FITS can serialize
+
+    Parameters
+    ----------
+    key : str
+        Key to be inserted in the dictionary
+    value : str or None
+        Value to be inserted
+
+    Notes
+    -----
+    This addresses a shortcoming of the FITS standard. There are length
+        restrictions on both the ``key`` (8 characters) and ``value`` (72
+        characters) in the FITS standard. There is a convention for handling
+        long keywords and a convention for handling long values, but the
+        two conventions cannot be used at the same time.
+
+        This addresses that case by checking the length of the ``key`` and
+        ``value`` and, if necessary, shortening the key.
+    Reference
+    ---------
+    This helper method is taken from `astropy.nddata.cddata`. Not imported
+    as this is a private method, subject to change
+
+    """
+
+    if len(key) > 8 and len(value) > 72:
+        short_name = key[:8]
+        meta['HIERARCH {0}'.format(key.upper())] = (short_name, "Shortened name for {}".format(key))
+        meta['short_name'] = value
+    else:
+        meta[key] = value
+
+
+class NDCubeIOMixin(NDIOMixin):
+
+    # Inherit docstring from parent class
+    __doc__ = NDIOMixin.__doc__
+
+    MISSING = 'MISNG{0}'
+    EXTRA_COORDS = 'EXTCR{0}'
```

Comment:
```EXTRA_COORDS``` is getting to be an overloaded variable name.  Suggest changing it to something more explicit, like ```EXTRA_COORDS_FITS_LABEL```.  Probably something shorter than this but still explicit would be better.

Reply:
Okay, I would change to something else